### PR TITLE
Documentation: Update developer guide with EditRemarkCommand

### LINKED ISSUE #255: Add EditRemark to DG's Implementation Section

### PR DESCRIPTION
**Pull request by cheeggered**
Closes #255

### CODE REVIEW

File: docs/diagrams/EditXYZRemarkActivityDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+end note
+:MediConnectParser parses the command;
+:EditXYZRemarkCommandParser parses the arguments provided;
+if () then ([command is valid])
+  :New EditXYZRemarkCommand instance is created;
+  :XYZRemark item is added to XYZ with the specified index;
+  :The existing XYZ in database is replaced by the
+  edited XYZ and FilteredXYZList is updated;
+else ([command is invalid])
+  :Parser throws error message;
```
Comment (derekjxtan) [inline docs/diagrams/EditXYZRemarkActivityDiagram.puml:17]: Parser throws exception, instead of error message

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -401,6 +401,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+Given below is an example usage scenario and how the add `Appointment Remark` mechanism behaves at each step.
+
+Step 1. The user launches the application. The `Database` will be initialized with all data in the order that it was stored in.
+
+Step 2. The user inputs `remark_a 2 r\follow up required` to add a remark to the second appointment in the appointment list.
+The `remark_a` command calls `AppointmentRemarkCommandParser#parse` which parses the parameters that build the remark to be added.
+A new `AppointmentRemarkCommand` instance will be created with the correct `Remark` object to be added.
+
+Step 3. The created `AppointmentRemarkCommand` instance is returned to `LogicManager` and its `execute` method is called.
+`AppointmentRemarkCommand#execute` then calls `Model#setAppointment` and with the given `Appointment Remark`.
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:423]: `Appointment Remark` change to `Remark`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -401,6 +401,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+This section describes the appointment/doctor/patient remark features.
+
+#### Implementation
+The adding/deleting/editing of a remark for an appointment/doctor/patient in MediConnect is facilitated by 'LogicManager'. It extends 'Logic' and stores the mediConnectParser that parses the user input, and the model in which the command is executed. Additionally, it implements the following operations:
+
+* `LogicManager#execute()` —  Executes the given user String input and returns a 'CommandResult'
+
+These operations are exposed in the `Ui` interface as `Ui#executeCommand()`.
+
+Given below is an example usage scenario and how the add `Appointment Remark` mechanism behaves at each step.
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:414]: "add `Appointment Remark`" change to "add `Remark` into an `Appointment`"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -401,6 +401,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+
+Step 2. The user inputs `remark_a 2 r\follow up required` to add a remark to the second appointment in the appointment list.
+The `remark_a` command calls `AppointmentRemarkCommandParser#parse` which parses the parameters that build the remark to be added.
+A new `AppointmentRemarkCommand` instance will be created with the correct `Remark` object to be added.
+
+Step 3. The created `AppointmentRemarkCommand` instance is returned to `LogicManager` and its `execute` method is called.
+`AppointmentRemarkCommand#execute` then calls `Model#setAppointment` and with the given `Appointment Remark`.
+The edited `Appointment` is then added to the filteredAppointmentList by calling `FilteredList#setAppointment`.
+
+The example usage scenario for the add patient remark and add doctor remark mechanisms would be very similar to the above scenario.
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:426]: should we generalise "add" to be "edit" instead? if yes, reflect the changes throughout

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -401,6 +401,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+
+The following sequence diagram shows how the add appointment operation would work:
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:428]: `add appointment` change to `add remarks`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -401,6 +401,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+![EditRemarkSequenceDiagram](images/EditRemarkSequenceDiagram.png)
+
+The following activity diagram summarizes what happens when a user wants to add a new appointment/patient/doctor:
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:431]: `add a new appointment/patient/doctor` change to `edit a remark for appointment/patient/doctor`

**Review by yanghengtang — APPROVED** [review]

**Pull request merged**